### DISCUSSION TITLE
Save stream from socket.io stream package to the file

### DISCUSSION THREAD

**Discussion by CapLek** [jimmywarting/StreamSaver.js#279] [Q&A] [answered]
In the browser I'm requesting pretty large binary file (~1 Gb) from a server through socket.io stream. It's being streamed by chunks. Is it possible to save it to the file directly not via saving to Blob first? 

I've tried it this way and in it didn't work.

```js
const socketStream = reqiure('socket.io-stream')

const fileStream = streamSaver.createWriteStream('filename.zip')
const targetStream = socketStream.createsStream() // Readable-writable stream
socketStream(socket).emit('get-large-file', targetStream)
targetStream.pipe(fileStream)
```

Maybe I'm missing something? Thank you.

**Comment by jimmywarting** [comment, marked as answer]
If you have the possibility to save the file using nothing but the server then that would be my recommendation for solving things instead of using streamsaver to emulate what a server dose using service worker to get around the problem. streamsaver and other alike is mostly build for site that generates huge data on the client side (like a webcam recording)

anyhow... it looks like the stream is a NodeJS stream and not a whatwg (web) stream so `.pipe()` don't work together web streams uses pipeTo

so i bet u would have to read the stream and write manually to the StreamSaver 

```js
const stream = streamSaver.createWriteStream('filename.zip')
const writer = stream.getWriter()
writer.write(new Uint8Array(data))
```

**Reply by CapLek** [reply]
Thank you for reply. Write operation can only be done after all stream content is fetched?

**Reply by jimmywarting** [reply]
no, the hole point of streamsaver is to write chunks sequentially and piping data to to the drive :P 

node streams are async iterable and yields Buffers that are instances of Uint8Array, so you might be able to do something like:

```js
const stream = streamSaver.createWriteStream('filename.zip')
const writer = stream.getWriter()
for await (const chunk of targetStream) {
  writer.write(chunk)
}
writer.close()
```

**Reply by CapLek** [reply]
It worked, thanks a lot!
